File: App/Server/Models/Measurement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AspCoreServer.Models
{

    public enum MeasurementType
    {
        Undefined,
        Zeitreihe,
        Ortskurve,
        Sonstige
    }

    public class Measurement
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }

        public string FileName { get; set; }
        public string FilePath { get; set; }
        public long FileSize { get; set; }
        public MeasurementType MeasurementType { get; set; }
        public ICollection<RawMeasurement> RawMeasurements { get; set; }

        public Battery Battery { get; set; }
        public Stack Stack { get; set; }
        public Cell Cell { get; set; }

        public DateTime Created { get; set; }
        public DateTime Modified { get; set; }
        public DateTime Measured { get; set; }

        //Setting Default value
        public Measurement()
        {
            Name = "Name";
            Description = "Beschreibung";
            Created = DateTime.Now.ToUniversalTime();
            Modified = DateTime.Now.ToUniversalTime();
            Measured = DateTime.Now.ToUniversalTime();
        }
    }

}

File: App/Client/app/models/Measurement.ts
import { Battery } from './Battery';
import { Stack } from './Stack';
import { Cell } from './Cell';
import { RawMeasurement } from './RawMeasurement';

export enum MeasurementType {
    Undefined,
    Zeitreihe,
    Ortskurve,
    Sonstige
}

export interface Measurement {
    id: number;
    name: string;
    description: string;

    fileName: string;
    fileSize: number;
    measurementType: MeasurementType;
    rawMeasurements: RawMeasurement[];

    battery: Battery;
    stack: Stack;
    cell: Cell;

    created: Date;
    modified: Date;
    measured: Date;
}
